test/cpp/ext/otel/otel_plugin_test.cc

```diff
@@ -1281,7 +1282,17 @@ TEST_F(OpenTelemetryPluginOptionEnd2EndTest,
   EXPECT_EQ(absl::get<std::string>(server_attributes.at("key5")), "value5");
 }
 
-using OpenTelemetryPluginNPCMetricsTest = OpenTelemetryPluginEnd2EndTest;
+class OpenTelemetryPluginNPCMetricsTest
+    : public OpenTelemetryPluginEnd2EndTest {
+ protected:
+  void TearDown() override {
+    // We are tearing down OpenTelemetryPluginEnd2EndTest first to ensure that
+    // gRPC has shutdown before we reset the instruments registry.
+    OpenTelemetryPluginEnd2EndTest::TearDown();
+    grpc_core::GlobalInstrumentsRegistryTestPeer::
+        ResetGlobalInstrumentsRegistry();
```

Comment:
Thanks for fixing this! Though this means we need to keep the `OpenTelemetryPluginNPCMetricsTest`s to be the last to run.

Reply:
Only if the OTel plugin tests the globally registered metrics.. which we probably shouldn't, or if we need to, we can register them again.